### PR TITLE
Trivial: Improving error reporting when assert_has_calls fails

### PR DESCRIPTION
**Pull request by davidair**

### CODE REVIEW

**Comment by davidair** [comment]
@gpshead

**Comment by gpshead** [comment]
I pinged @1st1 for a second pair of eyeballs and to ponder if there are other places we should improve the assertion error messages.  This is a trivial change, not really worthy of a BPO issue.  A news entry can be added if we accept this.

File: Lib/unittest/mock.py [hunk 1]
```diff
@@ -862,7 +862,8 @@ def assert_has_calls(self, calls, any_order=False):
                 not_found.append(kall)
         if not_found:
             raise AssertionError(
-                '%r not all found in call list' % (tuple(not_found),)
+                '%r not all found in call list: %r' % (tuple(not_found),
+                    all_calls)
```
Comment (1st1) [inline Lib/unittest/mock.py]: I'd also include `self._mock_name` in the error message, like in https://github.com/python/cpython/pull/8205/files#diff-ff75b1b83c21770847ade91fa5bb2525R829

Comment (davidair) [reply Lib/unittest/mock.py]: What should the message look like? Would it be something along the lines of "%r does not contain all of %r in its call list, found %r instead", where the first format parameter would be self._mock_name or 'mock'?

Comment (gpshead) [reply Lib/unittest/mock.py]: I believe so, yes.  I just committed and edit from the github UI doing that.  awaiting CI test results.

File: Lib/unittest/mock.py [hunk 2]
```diff
@@ -862,7 +862,9 @@ def assert_has_calls(self, calls, any_order=False):
                 not_found.append(kall)
         if not_found:
             raise AssertionError(
-                '%r not all found in call list' % (tuple(not_found),)
+                '%r does not contain all of %r in its call list, '
+                'found %r instead' % (self._mock_name or 'mock',
+                                      tuple(not_found), all_calls)
```
Comment (dimaqq) [inline Lib/unittest/mock.py:867]: Not sure about `tuple` here, given that both `not_found` and `all_calls` are lists as evident from lines 855, 857.

Comment (gpshead) [reply Lib/unittest/mock.py:867]: While this change preserved the behavior of the existing code.  I don't really think it matters, the error message will be understandable regardless.

**Pull request merged**